gateway-server/src/main/java/org/apache/knox/gateway/GatewayServlet.java

```diff
@@ -277,4 +278,34 @@ public Enumeration<String> getInitParameterNames() {
       return config.getInitParameterNames();
     }
   }
+
+  private Exception sanitizeException(Exception e) {
+    if (e == null || e.getMessage() == null) {
+      return e;
+    }
+    if (!isErrorMessageSanitizationEnabled || e.getMessage() == null) {
+      return e;
+    }
+    String sanitizedMessage = e.getMessage().replaceAll("\\b(?:\\d{1,3}\\.){3}\\d{1,3}\\b", "[hidden]");
+    return createNewException(e, sanitizedMessage);
+  }
+
+  private <T extends Exception> T createNewException(T e, String sanitizedMessage) {
```

Comment:
I would prefer this be named something like createSanitizedException, and encapsulate the sanitization as part of the creation of the new object. Is there any scenario where the message would be sanitized and not result in a new corresponding Exception object?

Reply:
@pzampino I renamed the method. By encapsulation I assume you mean wrapping it in the proposed `SanitizedException` class?

Reply:
Actually, I mean modifying the error message within this same method that creates the new Exception, rather than passing the modified message as an argument to this method.